src/common/ProfileMenu.tsx

```diff
@@ -0,0 +1,125 @@
+import { config } from '@config'
+import {
+  Divider,
+  ListItemIcon,
+  Menu,
+  MenuItem,
+  Stack,
+  Tooltip,
+  Typography,
+  useTheme,
+} from '@mui/material'
+import { useMemo, useState } from 'react'
+import { redirect } from 'react-router-dom'
+
+import { UserAvatar } from '@/common'
+import { BusEvents, RoutePaths } from '@/enums'
+import { bus } from '@/helpers'
+import { useAuth, useMetamaskZkpSnapContext } from '@/hooks'
+import { UiIcon } from '@/ui'
+
+interface ProfileMenuProps {
+  anchorEl?: null | HTMLElement | undefined
+  handleClose: () => void
+}
+
+const MAX_LENGTH_DID = 12
+
+export default function ProfileMenu({ anchorEl, handleClose }: ProfileMenuProps) {
+  const [isCopied, setIsCopied] = useState(false)
+
+  const open = Boolean(anchorEl)
+  const { palette, spacing, shadows } = useTheme()
+  const { userDid } = useMetamaskZkpSnapContext()
+  const { logout } = useAuth()
+
+  let timeout: NodeJS.Timeout
+
+  const copyToClipboard = async () => {
+    try {
+      clearTimeout(timeout)
+      await navigator.clipboard.writeText(`${userDid}`)
+      setIsCopied(true)
+      timeout = setTimeout(() => {
+        setIsCopied(false)
+      }, 3000)
+    } catch (e) {
+      bus.emit(BusEvents['error'], 'Not copied, pleas try again')
+    }
+  }
+
+  const redirectToSupport = () => {
+    window.open(config.SUPPORT_LINK, '_blank', 'noopener noreferrer')
+  }
+
+  const logOut = async () => {
+    await logout()
+    redirect(RoutePaths.SignIn)
+  }
+
+  const userDidLabel = useMemo(() => {
+    if (userDid.length > MAX_LENGTH_DID) {
+      return userDid.substring(0, 8) + '...' + userDid.substring(userDid.length - 4)
+    }
+    return userDid
+  }, [userDid])
+
+  return (
+    <Menu
+      anchorEl={anchorEl}
+      id='account-menu'
+      open={open}
+      onClose={handleClose}
+      anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
+      transformOrigin={{ vertical: 'bottom', horizontal: 'center' }}
+      MenuListProps={{
+        sx: {
+          width: spacing(60.5),
```

Comment:
let's use just 60, probably mistake in the design

Reply:
fixed